### ISSUE TITLE
assert_enqueued_emails does not work when sending a parameterized email

### ISSUE THREAD

**Issue by johnloringpollard** [rails/rails#37605]
### Steps to reproduce
When I run `assert_enqueued_emails` I noticed it only works with `ActionMailer::DeliveryJob` and does not work with a `ActionMailer::Parameterized::DeliveryJob`. 

Does work
```
assert_enqueued_emails 1 do
  SubscriptionMailer.welcome.deliver_later
end
```

Does not work
```
assert_enqueued_emails 1 do
  SubscriptionMailer.with(User.first).welcome.deliver_later
end
```
So it appears the `assert_enqueued_emails` only works when you aren't sending a parameterized email.

### Expected behavior

This should result in a passed test
```.```

### Actual behavior
Running the above results in a failure
```1 jobs expected, but 0 were enqueued.```

### System configuration
**Rails version**:
Rails 6.0.0
**Ruby version**:
ruby 2.6.2p47

I believe the issue may be from the following line. When `delivery_job` is called on i.e. `SubscriptionMailer` it will only returns `ActionMailer::DeliveryJob` and not `ActionMailer::Parameterized::DeliveryJob`. I believe it needs to check for both?
https://github.com/rails/rails/blob/66cabeda2c46c582d19738e1318be8d59584cc5b/actionmailer/lib/action_mailer/test_helper.rb#L159

**Comment by Edouard-chin** [comment]
Yep we should put back this line https://github.com/rails/rails/commit/e139a3ce1376e1b74e0996f6f0ae17e584d5acad#diff-b59d1a42c20115a6405dda3d7f38143fR163

It got removed because we tried to introduce a new Delivery job responsible to deliver both regular mails and parameterized mails, but it was semi reverted in https://github.com/rails/rails/pull/34591
Until 6.1 get released we should put back the line. cc/ @gmcgibbon

**Comment by bogdanvlviv** [comment]
@johnloringpollard It'll be fixed in Rails 6.0.1. Thank you for reporting this.
